edx_exams/apps/api/v1/views.py

```diff
@@ -179,3 +180,53 @@ def patch(self, request, course_id):
             data = {"detail": "Invalid data", "errors": serializer.errors}
 
         return Response(status=response_status, data=data)
+
+
+class CourseExamConfigurationsView(APIView):
+    """
+    View to create and update course exam configs for a specific course.
+
+    Given a course id and a proctoring provider name, this view will either create a new course exam configuration
+    (if one doesn't exist), or modify the proctoring provider on an existing course exam config.
+
+    HTTP PATCH
+    Creates or updates a CourseExamConfiguration.
+    Expected PATCH data: {
+        'provider': 'test_provider',
+    }
+    **PATCH data Parameters**
+        * name: This is the name of the proctoring provider.
+    **Exceptions**
+        * HTTP_400_BAD_REQUEST
+    """
+
+    authentication_classes = (JwtAuthentication,)
+    permission_classes = (StaffUserPermissions,)
+
+    @classmethod
+    def handle_config(cls, provider, course_id):
+        """
+        Helper method that decides whether to update existing or create new config.
+        """
+        CourseExamConfiguration.objects.update_or_create(
+            course_id=course_id,
+            defaults={'provider': provider})
+        log.info(f"Created or updated course exam configuration course_id={course_id},provider={provider.name}")
+
+    def patch(self, request, course_id):
+        """
+        Create/update course exam configuration.
+        """
+        try:
+            provider = ProctoringProvider.objects.get(name=request.data['provider'])
+            self.handle_config(provider, course_id)
+            response_status = status.HTTP_200_OK
+            data = {}
+        except KeyError:
```

Comment:
Does the way I've handled the errors make sense here? Let me know if additional info is needed or if there is a better approach.

Reply:
I think this is good. You could also just proactively check for this value at the start of the view. 

```if not request.data.get('provider'):
         response_status = status.HTTP_400_BAD_REQUEST
         data = {"detail": "No proctoring provider name in request."}
```

The `get` will return `None` if `provider` isn't in `request.data`.

The difference is small. I'd say the advantage of doing it this way is that it's easier to read. It took me a second to figure out why you were handling a `KeyError`. I had to read the response data and scroll back up to see what could throw a `KeyError`. It's not a big deal, though. I think this is good.

You could do it in a serializer, as an alternative, but that might be overkill for something like this.

Reply:
@MichaelRoytman, I like this approach! My thought with the `KeyError` was having both errors handled in the try except block but I agree that that approach is less readable. I've gone ahead and updated.